herd/ASLSem.ml

```diff
@@ -385,6 +392,19 @@ module Make (C : Config) = struct
     (* Primitives and helpers                                                 *)
     (**************************************************************************)
 
+    let vbool b =  V.Val (Constant.Concrete (ASLScalar.S_Bool b))
+
+    let checkprop m_prop =
+      let* vprop = m_prop in
+      M.assign (vbool true) vprop >>! []
+
+    let somebool _ =
+      let v = V.fresh_var () in
+      let mbool b =
+        let vb = V.Val (Constant.Concrete (ASLScalar.S_Bool b)) in
+        M.assign v vb >>= fun () -> M.unitT vb  in
+      choice (M.unitT v) (mbool true) (mbool false)
```

Comment:
Is it exactly what the ASL syntax `UNKNOWN: boolean` does?

Reply:
No idea!

Reply:
After checking, in ASLSem, for the moment, `UNKNOWN: _t` produces a fresh variable. As you have a condition just in front of it, it will create an altT with the two possibilities.

Reply:
Do you mean that we can write `SomeBoolean()` using `UNKNOW`? Something like
```
func SomeBoolean() => boolean
begin
  return if UNKNOWN:boolean then TRUE else FALSE end;
end
```
(Have not tried).

Reply:
This would indeed have the same effect as writing it as a primitive.

But actually, what I was suggesting is not to implement any function `SomeBool` and just use `UNKNOWN: boolean` instead of calling the function.

Reply:
This is doable for `UNKNOWN:boolean` but not for `UNKNOWN:integer`. So we would have a significant discrepancy of semantics. This would be disturbing for users, or?

Reply:
I don't understand the parallel with integers, we are not implementing `SomeInteger` here?

For clarification I am not suggesting to change the semantics of `UNKNOWN: boolean` to match those of `SomeBoolean`.

Reply:
I have tried to replace the call to `SomeBoolean()` by:
```
SuccessVA =
    if UNKNOWN:boolean then TRUE else FALSE;
```
It does not work. Lacking time to experiment, I keep the current solution with a specific `SomeBoolean()` primitive. Please do not hesitate to post a follwoup, even after merging, which is imminent.

Reply:
Okay, a shame but not too costly. It'd be good to investigate why using `UNKNOWN:boolean` does not work there.